src/drt/src/io/GuideProcessor.h

```diff
@@ -208,4 +235,199 @@ class GuideProcessor
   std::map<frNet*, std::vector<frRect>, frBlockObjectComp> tmp_guides_;
   std::vector<std::pair<frBlockObject*, Point>> tmpGRPins_;
 };
+
+/**
+ * @brief A class for traversing guides graph to check pin coverage and minimize
+ * guides.
+ *
+ * The GuidePathFinder class helps in finding paths between guides and pins on a
+ * net. It works by constructing adjacency lists of guides
+ * and pins, and then traverses the graph to find the minimal set of guides that
+ * visit all pins. It commits the found path to the final guides of the net.
+ */
+class GuidePathFinder
+{
+ public:
+  GuidePathFinder(frDesign* design, Logger* logger, frNet* net)
+      : design_(design), logger_(logger), net_(net)
+  {
+  }
+  int getNodeCount() const { return node_count_; }
+  int getGuideCount() const { return guide_count_; }
+  int getPinCount() const { return node_count_ - guide_count_; }
+  bool isForceFeedThrough() const { return force_feed_through_; }
+  bool allowWarnings() const { return allow_warnings_; }
+  /**
+   * @brief Initializes the node map and other member variables.
+   *
+   * The function constructs the node map from the given list of guides and
+   * pins.
+   *
+   * @param rects A vector of guide rectangles(by GCell indices).
+   * @param pin_gcell_map A map of pins and their corresponding GCell indices.
+   */
+  void init(const std::vector<frRect>& rects,
+            const frBlockObjectMap<std::set<Point3D>>& pin_gcell_map);
```

Comment:
The initialization is different at each iteration because the guides are split differently which leads to a different node map. I guess I could create a new GuidePathFinder for each iteration instead of having one and clearing it.